nexuse2e-webapp/src/main/webapp/WEB-INF/pages/reporting/statistics.jsp

```diff
@@ -24,57 +24,262 @@
 <%@ taglib uri="/tags/struts-nested" prefix="nested"%>
 <%@ taglib uri="/tags/struts-tiles" prefix="tiles"%>
 <%@ taglib uri="/tags/struts-logic" prefix="logic"%>
-<%@ taglib uri="http://java.sun.com/jstl/core" prefix="c"%>
+<%@ taglib uri='http://java.sun.com/jsp/jstl/core' prefix='c'%>
 <%@ taglib uri="/tags/nexus" prefix="nexus"%>
 
 <% /*<nexus:helpBar helpDoc="documentation/NEXUSe2e.html" /> */ %>
 
-<table class="NEXUS_TABLE" width="100%">
+<table class="NEXUS_TABLE" style="width: 100%">
 	<tr>
 		<td><nexus:crumbs /></td>
 	</tr>
 	<tr>
-		<td class="NEXUSScreenName">NEXUSe2e</td>
+		<td class="NEXUSScreenName">Dashboard</td>
 	</tr>
 </table>
 
-<logic:greaterThan name="messageCount" value="0">
-	<nexus:reportsAvailable>
-		<table width="100%">
+<div class="statistics">
+	<h2>Conversations (last 24h)</h2>
+	<c:choose>
+		<c:when test="${not empty conversationStatusCounts}">
+			<div class="chart" id="conversations">
+				<logic:iterate id="statusCount" name="conversationStatusCounts">
+					<c:set var="percentage" value="${statusCount.value / conversationStatusTotal * 100}" />
+					<c:if test="${percentage > 0}">
+						<div class="segment"
+							 style="width: ${percentage}%; background: ${colors[statusCount.key]};"
+							 title="${statusCount.key} ${statusCount.value}">
+							<span class="status-name">
+								${statusCount.key}
+							</span>
+							<span class="count-label">
+								${statusCount.value}
+							</span>
+						</div>
+					</c:if>
+				</logic:iterate>
+			</div>
+		</c:when>
+		<c:otherwise>
+			<p class="no-data">no conversations</p>
+		</c:otherwise>
+	</c:choose>
+
+	<h2>Failed Messages</h2>
+	<c:choose>
+	<c:when test="${not empty messages}">
+		<table class="NEXUS_TABLE fixed-table">
+			<colgroup>
+				<col>
+				<col style="width: 6%">
+				<col style="width: 10%">
+				<col style="width: 30%">
+				<col style="width: 30%">
+				<col style="width: 78px">
+			</colgroup>
 			<tr>
-				<td>
-					<nexus:report name="message_stati_24h">
-						<nexus:reportParam name="startDate" value="${last24Hours}"/>
-					</nexus:report>
-				</td>
-				<td>
-					<nexus:report name="conversation_stati_24h">
-						<nexus:reportParam name="startDate" value="${last24Hours}"/>
-					</nexus:report>
-				</td>
+				<th class="NEXUSSection">Time</th>
+				<th class="NEXUSSection">Partner</th>
+				<th class="NEXUSSection">Choreography</th>
+				<th class="NEXUSSection">Conversation</th>
+				<th class="NEXUSSection">Message</th>
+				<th class="NEXUSSection"></th>
 			</tr>
+			<logic:iterate id="message" name="messages">
+				<tr class="${message.messageId}">
```

Comment:
Using the messageId as directly css class looks to me strange. Maybe something like "message-${message.messageId}"